qiskit_nature/properties/second_quantization/electronic/integrals/electronic_integrals.py

```diff
@@ -80,8 +80,12 @@ def __str__(self) -> str:
             string += self._render_matrix_as_sparse_list(self._matrices)
         else:
             for title, mat in zip(self._matrix_representations, self._matrices):
+                rendered_matrix = self._render_matrix_as_sparse_list(mat)
                 string += [f"\t{title}"]
-                string += self._render_matrix_as_sparse_list(mat)
+                if not rendered_matrix:
+                    string[-1] += " is all zero"
```

Comment:
Is it ever possible that string is []?

Reply:
No this is not possible. See 2 lines above where I do `string += [f"\t{title}"]`. This ensures that `string[-1]` is always existent.